src/serializer/ResidualHeapVisitor.js

```diff
@@ -276,7 +277,8 @@ export class ResidualHeapVisitor {
   visitValueArray(val: ObjectValue): void {
     this.visitObjectProperties(val);
     const realm = this.realm;
-    let lenProperty = Get(realm, val, "length");
+    let lenProperty = getObjectPropertyInternalValue(val, "length");
```

Comment:
Can you explain the logic behind this change? `Get` is a well-defined operation so I can refer to the spec to learn what it does. How does `getObjectPropertyInternalValue` differ from it? Why is it important to use one but not the other here?

Reply:
The internal Get I created doesn’t return an AbstractValue if the object has leaked. The Get version does via OrdinaryGetOwnProperty in properties.js.

Reply:
Can you clarify more about why it's important to (not?) return AbstractValue and why that solves the issue we were seeing? We can chat tomorrow in person if you prefer.

Reply:
It’s because the “length” property on arrays is special and in most cases doesn’t need to be visited or serialized if it hasn’t been explicitly set. I’ll explain more tomorrow.